Break in v0.13->v0.14

Studying [this piece of code](https://github.com/phip1611/audio-visualizer/blob/2545d1a8404615cf4e8a38eb1d1ee6b89e638659/src/dynamic/live_input.rs#L144) from `audio-visualizer`, I extracted it in my play project as

```Rust
pub fn setup_stream(config: &StreamConfig, device: &Device, buffer: Arc<Mutex<AllocRingBuffer<f32>>>) -> Stream {
    device
        .build_input_stream(
            config,
            move |data: &[f32], _info| {
                let mut audio_buffer = buffer.lock().unwrap();
                audio_buffer.extend(data.iter().copied());
            },
            move |error: StreamError| eprintln!("error on input audio stream: {}", error),
        None
        )
        .unwrap()
}
```

Build is successful with ringbuffer `v0.13` but it fails with `v0.14`:

```
error[E0277]: `*mut f32` cannot be sent between threads safely
   --> src/lib.rs:24:13
    |
22  |           .build_input_stream(
    |            ------------------ required by a bound introduced by this call
23  |               config,
24  | /             move |data: &[f32], _info| {
25  | |                 let mut audio_buffer = buffer.lock().unwrap();
26  | |                 audio_buffer.extend(data.iter().copied());
27  | |             },
    | |_____________^ `*mut f32` cannot be sent between threads safely
    |
    = help: within `AllocRingBuffer<f32>`, the trait `Send` is not implemented for `*mut f32`
    = note: required because it appears within the type `AllocRingBuffer<f32>`
    = note: required for `Mutex<AllocRingBuffer<f32>>` to implement `Sync`
    = note: required for `Arc<Mutex<AllocRingBuffer<f32>>>` to implement `Send`
note: required because it's used within this closure
   --> src/lib.rs:24:13
    |
24  |             move |data: &[f32], _info| {
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^
note: required by a bound in `build_input_stream`
   --> ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cpal-0.15.2/src/traits.rs:134:46
    |
134 |         D: FnMut(&[T], &InputCallbackInfo) + Send + 'static,
    |                                              ^^^^ required by this bound in `DeviceTrait::build_input_stream`
```

There is no changelog or github tag to compare, but looking at the doc.rs pages for [0.13](https://docs.rs/ringbuffer/0.13.0/ringbuffer/struct.AllocRingBuffer.html) and [0.14](https://docs.rs/ringbuffer/0.14.0/ringbuffer/struct.AllocRingBuffer.html) what stands out is that it went from `Send` to `!Send`. Was that expected?

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "ringbuffer"
-version = "0.14.0"
+version = "0.14.1"
 authors = [
     "Victor Roest <victor@xirion.net>",
     "Jonathan Dönszelmann <jonabent@gmail.com>",

--- a/src/with_alloc/alloc_ringbuffer.rs
+++ b/src/with_alloc/alloc_ringbuffer.rs
@@ -86,6 +86,11 @@ pub struct AllocRingBuffer<T, SIZE: RingbufferSize = PowerOfTwo> {
     mode: PhantomData<SIZE>,
 }
 
+// SAFETY: all methods that require mutable access take &mut,
+// being send and sync was the old behavior but broke when we switched to *mut T.
+unsafe impl<T: Sync> Sync for AllocRingBuffer<T> {}
+unsafe impl<T: Send> Send for AllocRingBuffer<T> {}
+
 impl<T, const N: usize> From<[T; N]> for AllocRingBuffer<T, NonPowerOfTwo> {
     fn from(value: [T; N]) -> Self {
         let mut rb = Self::with_capacity_non_power_of_two(value.len());

